_infra/helm/locust/locustfiles/locustfile.py

```diff
@@ -18,80 +18,78 @@
+        "ce_events": "/mnt/locust/qbs_collection-exercise-event-config.json",
+        "type": "EQ",
+        "survey_ref": "139"
+    },
+    {
+        "survey_name": "ASHE",
+        "ce_config": "/mnt/locust/ashe_collection-exercise-config.json",
+        "ce_events": "/mnt/locust/ashe_collection-exercise-event-config.json",
+        "type": "SEFT",
+        "survey_ref": "141",
+        "ci_file_location": "/mnt/locust/065_201803_0001.xlsx"
+    },
+]
 
 # Load data for tests
 def load_data():
-    auth = (os.getenv('security_user_name'), os.getenv('security_user_password'))
-
-    logger.info("Container host: %s", socket.gethostname())
-
-    survey_id = load_survey(auth)
-    load_collection_exercises(auth)
-    load_collection_exercise_events(auth)
-    load_and_link_collection_instrument(auth, survey_id)
-    load_and_link_sample(auth)
-    execute_collection_exercise(auth, survey_id)
-    register_users(auth)
+    logger.info(f"Container host: {socket.gethostname()}")
 
+    for survey in SURVEY_DETAILS:
+        survey_id = get_survey_id(survey["survey_name"])
+        load_collection_exercises(survey["ce_config"])
+        load_collection_exercise_events(survey["ce_events"])
+        load_and_link_collection_instrument(survey_id, survey["type"], survey["survey_ref"],
+                                            survey.get("ci_file_location"))
+        load_and_link_sample(survey["survey_ref"])
+        execute_collection_exercise(survey["survey_ref"])
+    register_users()
 
-# Survey loading
-def load_survey(auth):
-    logger.info('Trying to find survey %s', survey_short_name)
-    get_url = f"{os.getenv('survey')}/surveys/shortname/{survey_short_name}"
-    get_response = requests.get(get_url, auth=auth)
 
+def get_survey_id(survey_name):
+    survey_details = requests.get(f"{os.getenv('survey')}/surveys/shortname/{survey_name}", auth=AUTH)
     try:
-        get_response.raise_for_status()
-        get_data = get_response.json()
-        logger.info("Survey successfully found at id %s", get_data['id'])
-        return get_data['id']
-    except requests.exceptions.HTTPError:
-        logger.error("Couldn't find survey %s, status code %s, message %s", survey_short_name, get_response.status_code,
-                     get_response.text)
-
-    create_url = f"{os.getenv('survey')}/surveys"
-    survey_details = {"surveyRef": survey_ref, "longName": survey_long_name, "shortName": survey_short_name,
-                      "legalBasisRef": 'STA1947', "surveyType": 'Business',
-                      "classifiers": [{"name": "COLLECTION_INSTRUMENT", "classifierTypes": ["FORM_TYPE"]},
-                                      {"name": "COMMUNICATION_TEMPLATE", "classifierTypes": ["LEGAL_BASIS", "REGION"]}]}
+        survey_details.raise_for_status()
+        survey_data = survey_details.json()
+        logger.info("Survey successfully found at id %s", survey_data["id"])
+        return survey_data["id"]
```

Comment:
I guess to guards against misconfiguration, but is this call to the survey service even needed? Is it not just the `survey_ref` in the `SURVEY_DETAILS` json?

Reply:
Not sure, there was so much stuff that needed updating and still is that I got to the stage of just getting it in. The registering of respondents is also another area which needs a lot of extra work